Added New NetLogo Work

smile4life opened the pull request:
This is my last working save - had just started to work on implementing
phloem and vascular bundle cells. The actual code is just in the first part and ends where I've commented "END" a bunch

...Does this mean you'll like us now? :grinning:

smile4life committed:
Added New NetLogo Work

This is my last working save - had just started to work on implementing
phloem and vascular bundle cells.

...Does this mean you'll like us now? :grinning:

alexanian commented on models/viralspread/netlogo/TryRodrigo.nlogo:
```diff
@@ -0,0 +1,664 @@
+end
+
+;; END OF ACTUAL CODE
+;; END
```
why does it say END so many times?

alexanian commented on models/viralspread/netlogo/TryRodrigo.nlogo:
```diff
@@ -0,0 +1,664 @@
+;; END
+;; END
+;; ...END?
+;; YES, END.
```
why does it say END so many times? Would be better to clarify what is below, "ACTUAL CODE" is a little confusing.

Also, is there any way to include the buttons etc. in a separate file and just import it? e.g. as discussed here http://stackoverflow.com/questions/5830045/including-a-netlogo-source-file-into-another

alexanian commented on models/viralspread/netlogo/TryRodrigo.nlogo:
```diff
@@ -0,0 +1,664 @@
+;;      tick
+;;    ]
+;;    stop  ;; stop model completely
+;;  ]
```
Not clear on the commented-out bit above here? Was that just to get a nice flat line at the end?

alexanian commented on models/viralspread/netlogo/TryRodrigo.nlogo:
```diff
@@ -0,0 +1,664 @@
+    [ 
+      ask link-neighbors with [not resistant?]
+        [ 
+          if (random-float 100 < viral-spread-chance) and (not infected?)
```
So basically, I think we now want some function like "viral-spread-from-neighbours" based on a cells-own "num-virions" paramter, rather than a random-float?

alexanian commented:
I like you anyway! I just don't have as clear idea of what you're working on without PRs. This is a nitpick, but it would be good to have a more specific commit message (e.g. "adds phloem links").

Overall well-commented and pretty easy to follow, though! Maybe make it more clear via comments that "to go" is the "main"  function that calls everything else? Some style of commenting that distinguishes the subroutines from the "to go"...

smile4life commented:
Yeah, the "END" was just to try to clearly delimit the contents of the NetLogo "Code" tab. I looked at the link, and I could do that, but it would be easier for me to just keep a separate .txt file without the interface and info text, if that's all right.

The commented-out bit was indeed just to get a nice flat line at the end. I left it in because I was thinking of giving the user the option of using that or specifying the number of "ticks"

Yeah, I think so :)

I can definitely do that for the future! I will also put proper comments in for the functions.

jrhawley commented:
Yeah, having a clear "end" to the code is nice to have, and I think I agree with what Tessa said. NetLogo has some strange syntax, but generally the code looks fine

alexanian commented:
@smile4life could you add another commit w/ the changes to comments (END + commented out chunk)? then I will merge. if it's much easier to have it all in the same file for now that's fine, we might just separate out for our code before the jamboree

smile4life committed:
Added phloem structure & virus count

Also tweaked some features of the cells & merged Robert's quadrant
separations

I'm not fully happy with the viral count implementation - still uses an
arbitrary random chance for the virions to actually move to neighbouring
cells.

I have a section commented out because I wasn't specific enough with
connecting the stem agents to the leaf cells

smile4life commented:
I _think_ I added another commit, but I'm not sure I can create a pull request while this one is still unresolved?

alexanian commented:
If you push your commit to the repo that your original pull request is
from, it will show up on the page.

smile4life commented:
Ah, and indeed it has - very cool :) Thanks!

alexanian commented:
LGTM, merging!
